PHP Task

kolawolejohn opened the pull request:
Submission for PHP task
- I added my sql code to the init.sql file to create my database
- to create the database I ran install.php
- once the database and the brands and products table have been created
- I used the create.php to add my products to the products table
- I checked if my products table have been populated via the overview.php
- I added some HTML and css file to the header.php and footer.php
- I was able to style my table and form

kolawolejohn committed:
This is my PHP Task

tegaphilip commented on submissions/kolawoleOgunfowokan/PHPTask/HTML/index.html:
```diff
@@ -0,0 +1,421 @@
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+    <link rel="stylesheet" href="../css/style.css">
```
change the href to `css/style.css`

kolawolejohn replied:
Done.

kolawolejohn replied:
Done, and it is fine.

tegaphilip commented on submissions/kolawoleOgunfowokan/PHPTask/php/create.php:
```diff
@@ -0,0 +1,61 @@
+        <?php session_start(); ?>
+
+        <select name="brand">
+            <?php for ($i = 0; $i < count($_SESSION['brands']); $i++) { ?>
```
Hey @kolawolejohn , I like the idea of using sessions to store the brands but from what I see, this value gets populated from `fetch.php` which is not loaded on the `create.php` page. So anytime this page gets loaded, the brands are empty. 

Maybe you should include the `fetch.php` page on this file as well.

kolawolejohn replied:
Fixed and quite good, thanks for pointing it out.

tegaphilip commented on submissions/kolawoleOgunfowokan/PHPTask/php/create.php:
```diff
@@ -0,0 +1,61 @@
+        $newProduct = array(
+            "itemname" => $_POST['itemname'],
+            "quantity" => $_POST['quantity'],
+            "brand_id" => $_POST['brand'] + 1
```
you are adding 1 here because you did not fetch the brand ids from the database in your `fetch.php` file. You should fetch the brand ids and brand names and create an associative array that maps ids to name and use that to build the `<select>` tag in your `create.php` file. That way, you don't need to add +1 anymore.

Think of it this way, what if brand item 3 is removed from your database. what effect will this your method of adding 1 have? problems, i suppose?

kolawolejohn replied:
This is the bone of contention. I couldn't fix this. So I still let the +1. I think I need help.

kolawolejohn replied:
I am through with this. Thanks @tegaphilip

tegaphilip reviewed (commented):
overall. great attempt. i actually cloned your repo and set up you php app on my PC. I created your database as well and it looks good. just run through the comments I made.

kolawolejohn committed:
I edited my CSS path, and I required fetch.php in my create.php file

kolawolejohn committed:
I recreated my products table

kenny-wealth commented on submissions/kolawoleOgunfowokan/PHPTask/php/fetch.php:
```diff
@@ -0,0 +1,19 @@
+    $stmt = $connection->query("SELECT * FROM myoja.brands");
+    $brands = array();
+    while ($item = $stmt->fetch()) {
+        array_push($brands, $item['brand_name']);
```
@kolawolejohn, map your brand_id to brand_name here, so you won't have to do +1 while saving the product.
You can replace array_push (line 14) with something like `$brands[$item['brand_id']] =  $item['brand_name'];`. 
This way you are assured that the each brand carries it's id in the database.
Your current approach will not work if your brand ids are not in order in the database.


To use the brands, you can easily iterate over the items in the array, since you know the sructure of your array.
So to load brands into the dropdown, you can have something like:
```
foreach ($brands as $id => $brandName) {
     <option value= <?php echo $id; ?>> <?php echo $brandName ?></option>
} 
```

kolawolejohn replied:
I am through with this and the necessary changes effected. One long ride init. "Foreach" saved a life. @kenny-wealth

kolawolejohn committed:
I worked on my fetch.php and create.php file

kolawolejohn committed:
I removed some unecessary space in my create.php file

kolawolejohn committed:
I edited my php file

tegaphilip reviewed (dismissed):
Looks good for a start. Approving. Hopefully, when we progress to using frameworks, you'll learn how to organize code much better. but great improvement from the previous version. 👍

JohnItoo commented on submissions/kolawoleOgunfowokan/PHPTask/php/overview_helper.php:
```diff
@@ -0,0 +1,17 @@
+        array_push($products, $item);
+    }
+    $_SESSION['products'] = $products;
+    var_dump($products);
```
Please remove this var_dump here.

kolawolejohn replied:
Done. It has been removed. Thanks.

kolawolejohn committed:
I removed the var_dump in my overview_helper.php file

JohnItoo commented on submissions/kolawoleOgunfowokan/PHPTask/php/overview.php:
```diff
@@ -0,0 +1,45 @@
+
+try {
+    $connection = new PDO("mysql:host=$host", $username, $password, $options);
+    $sql = file_get_contents("../data/fetch.sql");
```
You are not using this `$sql` variable anywhere please remove it.

JohnItoo commented on submissions/kolawoleOgunfowokan/PHPTask/php/overview.php:
```diff
@@ -0,0 +1,45 @@
+    while ($item = $stmt->fetch()) {
+        array_push($products, $item);
+    }
+    $_SESSION['products'] = $products;
```
Kindly remove this ` $_SESSION['products'] = $products;` it is not in use.

JohnItoo commented on submissions/kolawoleOgunfowokan/PHPTask/php/overview_helper.php:
```diff
@@ -0,0 +1,16 @@
+<?php
```
In Overview, you are fetching directly from the DB so you do not need this file. You can remove it

kolawolejohn committed:
I deleted uneccessary files and line of codes from my files

kolawolejohn commented:
I have attended to all the aforementioned conflicts. I deleted the overview_helper.php file, and removed some lines of code that were not needed in the overview.php file. @JohnItoo . Thanks

JohnItoo commented on submissions/kolawoleOgunfowokan/PHPTask/php/fetch.php:
```diff
@@ -0,0 +1,18 @@
+<?php
```
Hello. Please kindly remove this file.

JohnItoo commented on submissions/kolawoleOgunfowokan/PHPTask/php/create.php:
```diff
@@ -0,0 +1,50 @@
+    <label for="itemname">Name of Product</label>
+    <input type="text" name="itemname" id="itemname">
+    <select name="brand">
+        <?php foreach ($_SESSION['brands'] as $key => $value) { ?>
```
You should not use Sessions. You can always query your database when you are trying to render your form.

kolawolejohn committed:
I removed fetch.sql and also edited my foreach loop in the create.php file

kolawolejohn commented:
@JohnItoo I have removed the fetch.sql file, and not the fetch.php file as the fetch.php file was required while the fetch.sql was not.

Also I removed the $_SESSION['brands'] in my foreach loop in the create.php file and  all looks well now. Thanks

JohnItoo reviewed (approved):
Hello @kolawolejohn . I think this is a good starting point. Well done.

JohnItoo reviewed (changes requested):
Hello @kolawolejohn . Nice work. I mentioned during the physical session that your submission should be inside a `php-intro` directory and it should be agnostic of your entire work so far.  Kindly rename your directory from `php` to `php-intro` or something similar.

kolawolejohn committed:
I changed the name of my php task folder to php-intro, instead of just php

kolawolejohn reviewed (commented):
> Hello @kolawolejohn . Nice work. I mentioned during the physical session that your submission should be inside a `php-intro` directory and it should be agnostic of your entire work so far. Kindly rename your directory from `php` to `php-intro` or something similar.

This has been done sir. Thank you @JohnItoo

kolawolejohn committed:
Change the task directory to PHP-intro

kolawolejohn committed:
Merge branch 'master' into PHPTask
